src/renderer/register/actions/registerFormActions.js

```diff
@@ -39,14 +39,24 @@ export const validateAndStoreFormData = async (data) => {
     throw new Error('Fill in a Secret Word.');
   }
 
-  // Generate bip39 Mnemonic - 256-bits entropy (24-word long mnemonic)
-  const mnemonic = bip39.generateMnemonic(256, null, bip39.wordlists[DEFAULT_LANGUAGE]);
-  const encryptedMnemonic = simpleEncrypt(mnemonic, passphrase);
-  return {
-    ...data,
-    mnemonic,
-    encryptedMnemonic
-  };
+  if (!isHardware && !isImport) {
```

Comment:
You can safely adjust this if/else if/else as mnemonic is only NOT required on the first step when isImport is true 

Instead you could do something like this: 
```js
if(isImport && !!data.mnemonic) return data;

const mnemonic = isImport 
                                    ? data.mnemonic  
                                    : bip39.generateMnemonic(256, null, bip39.wordlists[DEFAULT_LANGUAGE]);

const encryptedMnemonic = simpleEncrypt(mnemonic, passphrase);
return {
     ...data,
    mnemonic.
    encryptedMnemonic
};
```
(not 100% if that's correct :P)

Reply:
This wouldn't work since data gets returned before encrypting the mnemonic

Reply:
I see your point though. Will try out something